lib/puppet_x/coi/jboss/provider/securitydomain.rb

```diff
@@ -1,35 +1,56 @@
 # A class for JBoss security domain provider
 module Puppet_X::Coi::Jboss::Provider::SecurityDomain
+
+  # Method that creates security-domain in Jboss instance. When invoked it will execute 3 commands, add cache-type with value 'default', add authentication with value classic, add login-modules. Depends on the version of server it will use correct path to set security domain
   def create
-    cmd = "/subsystem=security/security-domain=#{@resource[:name]}/authentication=classic:add(login-modules=[{code=>\"#{@resource[:code]}\",flag=>\"#{@resource[:codeflag]}\",module-options=>["
-    options = []
-    @resource[:moduleoptions].keys.sort.each do |key|
-      value = @resource[:moduleoptions][key]
-      val = value.to_s.gsub(/\n/, ' ').strip
-      options << '%s => "%s"' % [key, val]
-    end
-    cmd += options.join(',') + "]}])"
-    cmd = compilecmd(cmd)
+    commands_template = create_parametrized_cmd
+    commands = commands_template.join('/')
+
+    cmd = compilecmd commands
     cmd2 = compilecmd "/subsystem=security/security-domain=#{@resource[:name]}:add(cache-type=default)"
+
     bringUp('Security Domain Cache Type', cmd2)[:result]
+
+    # TODO: Implement some nice way to decide if this method should be invoked, simple if is bleeeh.
+    if not @auth
+      cmd3 = compilecmd "/subsystem=security/security-domain=#{@resource[:name]}/authentication=classic:add()"
+      bringUp('Security Domain Authentication', cmd3)[:result]
+    end
+
     bringUp('Security Domain', cmd)[:result]
   end
 
+  # Method to remove security-domain from Jboss instance
   def destroy
     cmd = compilecmd "/subsystem=security/security-domain=#{@resource[:name]}:remove()"
     bringDown('Security Domain', cmd)[:result]
   end
 
+  # Method to check if there is security domain. Method calls recursive read-resource on security subsystem to validate if security domain is present. In the procces method also checks if authentication is set.
   def exists?
-    cmd = compilecmd "/subsystem=security/security-domain=#{@resource[:name]}/authentication=classic:read-resource()"
+    cmd = compilecmd "/subsystem=security:read-resource(recursive=true)"
     res = executeWithoutRetry cmd
+
     if not res[:result]
       Puppet.debug "Security Domain does NOT exist"
       return false
```

Comment:
Alternatively to comment on line 15 comment (https://github.com/coi-gov-pl/puppet-jboss/pull/55/files#r58560122) You can use something like this here (instead of just `return false`):

```ruby
return saving_authentication_with false
```
and

```ruby
def saving_authentication_with(setting)
  @auth = if setting then true else false end
  @auth
end
```